src/callbacks/callbacks.jl

```diff
@@ -434,6 +434,64 @@ function save_restart_func(integrator, output_dir)
     return nothing
 end
 
+Base.@kwdef mutable struct WallTimeEstimate
+    n_calls::Int = 0
+    n_next::Int = 1
+    t_last::Float64 = -1
+    ∑Δt_wall::Float64 = 0
+end
+import Dates
+function print_walltime_estimate(integrator)
+    (; walltime_estimate, dt, t_end) = integrator.p
+    wte = walltime_estimate
+    # 1st call initializes `t_last` (during integrator init)
+    # 2nd call computes `Δt` for the first time (but also includes compilation), so we skip
+    cond = wte.n_calls > 1
+    if cond
```

Comment:
Can you please rename `cond` to something more descriptive? Maybe `is_first_invocation` and flip the conditionals?

Reply:
It's actually the _third_ invocation (`n_calls` is zero to start).

 - The very first call (when `n_calls == 0`), there's no elapsed times to report (and this is called during initialization, before `step!` has been called).
 - The second call (`n_calls == 1`) is after `step!` is called for the first time, but we don't want to report this since it includes compilation time.
 - Calls after that (`n_calls > 1`) exclude compilation and provide the best wall time estimates

Maybe we should include this list/description above in the docs?

Regarding the variable name, how about `ready_to_report = wte.n_calls > 1`? (that's what I'll do tentatively, let me know if you prefer something else)

Reply:
Please unresolve if you think it still needs another look